src/main/java/org/tron/core/net/node/NodeImpl.java

```diff
@@ -1280,21 +1287,27 @@ private void startSyncWithPeer(PeerConnection peer) {
   }
 
   private void syncNextBatchChainIds(PeerConnection peer) {
-    if (peer.getSyncChainRequested() != null) {
-      logger.info("Peer {} is in sync.", peer.getNode().getHost());
-      return;
-    }
-    try {
-      Deque<BlockId> chainSummary =
-          del.getBlockChainSummary(peer.getHeadBlockWeBothHave(),
-              peer.getSyncBlockToFetch());
-      peer.setSyncChainRequested(
-          new Pair<>(chainSummary, System.currentTimeMillis()));
-      peer.sendMessage(new SyncBlockChainMessage((LinkedList<BlockId>) chainSummary));
-    } catch (TronException e) {
-      logger.error("Peer {} sync next batch chainIds failed, error: {}.", peer.getNode().getHost(),
-          e.getMessage());
-      disconnectPeer(peer, ReasonCode.FORKED);
+    synchronized (peer){
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: "peer" is a method parameter, and should not be used for synchronization. ([squid:S2445](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2445))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1672&issues=AWbTi2UkLpqqSUlcbPrt&open=AWbTi2UkLpqqSUlcbPrt)

[ISSUE_KEY]: AWbTi2UkLpqqSUlcbPrt